### DISCUSSION TITLE
Are there alternative cache eviction policies?

### DISCUSSION THREAD

**Discussion by alexreinking** [ccache/ccache#614] [Questions & answers] [answered]
### Question ###
Does ccache track the amount of time it takes for a source file to compile? We have some CUs that take a long time to compile due to template metaprogramming and it would be nice to have their object files not be evicted as eagerly as the LRU policy might do it.

**Comment by jrosdahl** [comment, marked as answer]
No, ccache does not track the time spent for compilation but there have been such ideas floating around before; see #453. However, even if ccache would measure and store the CPU time it took to compile a file I don't think there is a good way of making use of it for cache eviction purposes given the current cache storage model where there is no central storage of metadata other than statistics counters. That is, to evict based on CPU time ccache would have to read the data from each individual cache entry (file) and that would be too costly.

**Comment by jrosdahl** [comment]
@alexreinking: Are you happy with the answer or are you looking for more feedback?
